code/modules/mob/living/carbon/human/species.dm

```diff
@@ -1411,9 +1411,84 @@ GLOBAL_LIST_EMPTY(roundstart_races)
 	if(user == target)
 		return FALSE
 	if(user.loc == target.loc)
+		if(user.pulling == target && user.zone_selected == BODY_ZONE_HEAD && user.grab_state >= GRAB_NECK)
+			start_noogie(user, target)
 		return FALSE
 	user.disarm(target)
 
+/// For starting a noogie, leads into the actual noogie loop
+/datum/species/proc/start_noogie(mob/living/carbon/human/user, mob/living/carbon/human/target)
+	if(!(target?.get_bodypart(BODY_ZONE_HEAD)) || user.pulling != target || user.getStaminaLoss() > 80)
+		return FALSE
+
+	var/obj/item/bodypart/head/the_head = target.get_bodypart(BODY_ZONE_HEAD)
+	if(!(HAS_FLESH in target.dna?.species.species_traits) || !the_head.is_organic_limb())
+		to_chat(user, "<span class='warning'>You can't noogie [target], [target.p_they()] have no skin on [target.p_their()] head!</span>")
+		return
+
+	// [user] gives [target] a [prefix_desc] noogie[affix_desc]!
+	var/prefix_desc = "rough"
+	var/affix_desc = ""
+	var/affix_desc_target = ""
+
+	if(HAS_TRAIT(target, TRAIT_ANTENNAE))
+		prefix_desc = "violent"
+		affix_desc = "on [target.p_their()] sensitive antennae"
+		affix_desc_target = "on your highly sensitive antennae"
+	if(user.dna?.check_mutation(HULK))
+		prefix_desc = "sickeningly brutal"
+
+	var/message_others = "[prefix_desc] noogie[affix_desc]"
+	var/message_target = "[prefix_desc] noogie[affix_desc_target]"
+
+	user.visible_message("<span class='danger'>[user] begins giving [target] a [message_others]!</span>", "<span class='warning'>You start giving [target] a [message_others]!</span>", vision_distance=COMBAT_MESSAGE_RANGE, ignored_mobs=target)
+	to_chat(target, "<span class='userdanger'>[user] starts giving you a [message_target]!</span>")
+
+	if(!do_after(user, 1.5 SECONDS, target))
+		to_chat(user, "<span class='warning'>You fail to give [target] a noogie!</span>")
+		to_chat(target, "<span class='danger'>[user] fails to give you a noogie!</span>")
+		return
+
+	if(prefix_desc == "rough")
+		SEND_SIGNAL(target, COMSIG_ADD_MOOD_EVENT, "noogie", /datum/mood_event/noogie)
+	else
+		SEND_SIGNAL(target, COMSIG_ADD_MOOD_EVENT, "noogie_harsh", /datum/mood_event/noogie_harsh)
+
+	noogie_loop(user, target, 0)
+
+/// The actual meat and bones of the noogie'ing
+/datum/species/proc/noogie_loop(mob/living/carbon/human/user, mob/living/carbon/human/target, iteration)
+	if(!(target?.get_bodypart(BODY_ZONE_HEAD)) || user.pulling != target)
+		return FALSE
+
+	if(user.getStaminaLoss() > 80)
+		to_chat(user, "<span class='warning'>You're too tired to continue giving [target] a noogie!</span>")
+		to_chat(target, "<span class='danger'>[user] is too tired to continue giving you a noogie!</span>")
+		return
+
+	var/damage = rand(1, 5)
+	if(HAS_TRAIT(target, TRAIT_ANTENNAE))
+		damage += rand(3,7)
+	if(user.dna?.check_mutation(HULK))
+		damage += rand(3,7)
+
+	if(damage >= 5)
+		target.emote("scream")
+
+	target.apply_damage(damage, BRUTE, BODY_ZONE_HEAD)
+	user.adjustStaminaLoss(iteration + 5)
```

Comment:
remember that taking stamina damage resets the amount of time remaining until your stamina damage will reset itself to 0, so I don't really think that making the stamina damage increase with each noogie is necessary.

if someone wants to take hard drugs just so that they can noogie someone indefinitely, then by god, we should let them.